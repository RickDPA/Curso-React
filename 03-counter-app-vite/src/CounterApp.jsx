import { useState } from 'react';
import PropTypes from 'prop-types';



export const CounterApp = ({value}) => {

    const [ counter, setCounter] = useState(value);

    const suma = () => {
        setCounter(counter + 1);
    }

    const resta = () => {
        setCounter(counter -1);
    }

    const resetear = () => {
        setCounter(value);
    }

    return (
        <>

            <h1>CounterApp</h1>
            <h2> { counter } </h2>

            <button onClick={suma}>+1</button>
            <button onClick={resta}>-1</button>
            <button aria-label="btn-reset" onClick={resetear} className="rojo"> Reset</button>

        </>
    )
}

CounterApp.propTypes = {
    value: PropTypes.number.isRequired
}
