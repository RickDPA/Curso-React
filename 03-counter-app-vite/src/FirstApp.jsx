import PropTypes from 'prop-types';

/*
const getResult = () => {
    return 4 + 4;
}
*/
export const FirstApp = ({title, subTitle, name}) => {

    //console.log(props);

    return (
        <>
            {/*<h1>{getResult()}</h1>
            <code>{JSON.stringify(newMessage)}</code>*/}
            <h1 data-testid="test-title">{title}</h1>
            <p>{subTitle}</p>
            <p>{name}</p>
        </>
    )
}

FirstApp.propTypes = {
    title: PropTypes.string.isRequired,
    subTitle: PropTypes.string
}

FirstApp.defaultProps = {
    title: "No hay titulo",
    subTitle: "No hay subtitulo",
    name: "Ricardo Penagos"
}