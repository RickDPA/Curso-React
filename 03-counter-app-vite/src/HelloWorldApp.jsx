/*
export function App(){
    //document.createElement...
    return <h1>Hola Mundo!!!</h1>;
}

export const hola = 123;
*/

export const HelloWorldApp = () => {
  return (
    <h1>HelloWorldApp!!</h1>
  )
}
