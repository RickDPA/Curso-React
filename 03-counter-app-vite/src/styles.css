html, body {
    background-color: #21232A;
    color: white;
    font-family: Helvetica, Arial, sans-serif;
    font-size: 1.3rem;
    padding: 70px;
}

button{
    padding: 15px;
    margin-left: 10px;
    border-color: white;
    border-inline-width: 2px;
    font-size: 1.3rem;
    cursor: pointer;
    color: white;
    background-color: blue;
}

.rojo{
    padding: 15px;
    margin-left: 10px;
    border-color: white;
    border-inline-width: 2px;
    font-size: 1.3rem;
    cursor: pointer;
    color: white;
    background-color: red; 
}