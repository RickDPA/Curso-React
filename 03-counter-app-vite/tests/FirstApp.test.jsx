import { render} from "@testing-library/react"
import { FirstApp } from "../src/FirstApp"


describe('Pruebas en FirstApp', () => { 
    
    /*
    test('Debe de hacer match con el snapshot', () => { 
        
        const title = 'Hola, soy Goku'
        const {container} = render(<FirstApp title={title}></FirstApp>);
        //console.log(container);

        expect(container).toMatchSnapshot();

    });
    */

     test('Debe mostrar el titulo h1', () => { 
        
        const title = 'Hola, soy Goku'
        const {container, getByText, getByTestId} = render(<FirstApp title={title}></FirstApp>);
        
        expect(getByText(title)).toBeTruthy();

        /*
        const h1 = container.querySelector('h1');
        expect(h1.innerHTML).toContain(title);
        */
        expect(getByTestId('test-title').innerHTML).toBe(title);

     })


     test('Debe mostrar el subtitulo enviado por props', () => { 
        
        const title = 'Hola, soy Goku';
        const subTitle = 'Soy un subtitulo';
        const {getAllByText} = render(
        <FirstApp 
            title={title}
            subTitle={subTitle}
        />
        );
        
        expect(getAllByText(subTitle).length).toBe(1);


     })
 })