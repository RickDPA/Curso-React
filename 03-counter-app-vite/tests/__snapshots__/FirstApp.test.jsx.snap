// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en FirstApp Debe de hacer match con el snapshot 1`] = `
<div>
  <h1>
    Hola, soy Goku
  </h1>
  <p>
    No hay subtitulo
  </p>
  <p>
    Ricardo Penagos
  </p>
</div>
`;

exports[`Pruebas en FirstApp Debe de hacer match con el snapshot 2`] = `
<div>
  <h1>
    Hola, soy Goku
  </h1>
  <p>
    No hay subtitulo
  </p>
  <p>
    Ricardo Penagos
  </p>
</div>
`;

exports[`Pruebas en FirstApp Debe mostrar el titulo h1 1`] = `
<div>
  <h1>
    Hola, soy Goku
  </h1>
  <p>
    No hay subtitulo
  </p>
  <p>
    Ricardo Penagos
  </p>
</div>
`;
